virtual/vagrant_startup.sh

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env bash
+
+. /etc/os-release
+
+# Install Software
+case "$ID_LIKE" in
+    rhel*)
+        # Install Vagrant
+	set -xe
+
+
+	# update yum
+	sudo yum  update
+
+	# install essential packages and tools
+	sudo yum -y install wget
+	sudo yum group install -y "Development Tools"
+	sudo yum install -y centos-release-qemu-ev qem-kvm-ev qemu-kvm libvirt virt-install bridge-utils libvirt-devel  libxslt-devel libxml2-devel libvirt-devel libguestfs-tools-c
+
+	# set up networking for Vagrant VMs
+	#sudo echo "net.ipv4.ip_forward = 1"|sudo tee /etc/sysctl.d/99-ipforward.conf
+	#sudo sysctl -p /etc/sysctl.d/99-ipforward.conf
+
+	# start up libvirt as our VM method for Vagrant
+	sudo usermod -a -G libvirt $(whoami)
+	sudo systemctl enable libvirtd
+	sudo systemctl start libvirtd
+
+
+	# install vagrant (frozen at 2.2.3 to avoid various issues)
+	pushd "$(mktemp -d)"
+	wget https://releases.hashicorp.com/vagrant/2.2.3/vagrant_2.2.3_x86_64.rpm -O vagrant.rpm
+	#sudo rpm -i vagrant.rpm
+	sudo yum -y localinstall vagrant.rpm
+	popd
+
+	# install other dependencies
+	sudo yum install -y sshpass
+
+	# install kvm packages
+	sudo yum install -y qemu-kvm libvirt-bin libvirt-dev bridge-utils libguestfs-tools
+	sudo yum install -y qemu virt-manager firewalld OVMF
+
+	# install vagrant plugins
+	vagrant plugin install vagrant-hostmanager vagrant-libvirt
+	vagrant plugin install vagrant-host-shell vagrant-scp vagrant-mutate
+
+	#set up Vagrantfile and start up the configuration in Vagrant
+	export DEEPOPS_VAGRANT_FILE="/home/$(whoami)/deepops/virtual/Vagrantfile-centos"
```

Comment:
1. This assumes the location of the deepops repo and home directory. Instead, maybe use the pattern for setting `VIRT_DIR` further down to get the absolute path of the script directory? 
1. This line (and the one like it for Ubuntu) assumes we want the same host OS and VM OS, but there are use cases where we'll want these to be different. Instead of hard-coding this, can we get it from the environment and set a default? E.g.,

```
DEEPOPS_VAGRANT_FILE=${DEEPOPS_VAGRANT_FILE:-${VIRT_DIR}/Vagrantfile-centos}
```

Reply:
Sure. I still think the logic below this line permits that but was making it a setting that would work OOB. Long term I talked to Doug and agree that the Vagrant file should be auto generated based on config files rather than be static. I think he was going to have a .yml file and ansible the appropriate Vagrantfile.